module/VuFind/src/VuFind/Resolver/Driver/Alma.php

```diff
@@ -50,18 +50,26 @@ class Alma extends AbstractBase
      *
      * @var array
      */
-    protected $ignoredFilterReasons = ['Date Filter'];
+    protected $ignoredFilterReasons;
 
     /**
      * Constructor
      *
-     * @param string               $baseUrl    Base URL for link resolver
-     * @param \Laminas\Http\Client $httpClient HTTP client
+     * @param string                 $baseUrl    Base URL for link resolver
+     * @param \Laminas\Http\Client   $httpClient HTTP client
+     * @param \Laminas\Config\Config $config     VuFind Configuration (optional)
      */
-    public function __construct($baseUrl, \Laminas\Http\Client $httpClient)
-    {
+    public function __construct(
+        $baseUrl,
+        \Laminas\Http\Client $httpClient,
+        \Laminas\Config\Config $config = null
+    ) {
         parent::__construct($baseUrl);
         $this->httpClient = $httpClient;
+        $this->ignoredFilterReasons = explode(
```

Comment:
What if you just accepted `array $options` instead of a Config object? You could convert the OpenURL Config section to an array in the factory, and then this class has one less dependency. (@EreMaijala and I have talked about reducing dependency on the Config class, since it's not as useful as it used to be, and if it should ever go out of support in the future, it will be that much less work to factor it out later).

My thought is that you could default the property to whatever value you like (either the old default, if we decide to keep it, or an empty array if we don't). Then the code here could be something like:

```
if (isset($options['ignoredFilterReasons'])) {
    $this->ignoredFilterReasons = empty($options['ignoredFilterReasons']) ? [] : (array)$options['ignoredFilterReasons'];
}
```

(Of course, that assumes you take my suggestions above about using array-type configuration settings and supporting false to disable filtering... if you prefer to stick with the delimited string, the code would look a little different).